### PR TITLE
feat(vm): Implement try-catch

### PR DESCRIPTION
**Pull request by andreubotella**
Do not merge before #181.

### CODE REVIEW

File: nova_vm/src/engine/bytecode/executable.rs [hunk 1]
```diff
@@ -1159,6 +1159,44 @@ impl CompileEvaluation for ast::ThrowStatement<'_> {
...
+        let jump_to_catch = ctx
+            .exe
+            .add_instruction_with_jump_slot(Instruction::PushExceptionJumpTarget);
+        self.block.compile(ctx);
+        ctx.exe.add_instruction(Instruction::PopExceptionJumpTarget);
+        let jump_to_end = ctx.exe.add_instruction_with_jump_slot(Instruction::Jump);
+
+        let catch_clause = self.handler.as_ref().unwrap();
+        ctx.exe.set_jump_target_here(jump_to_catch);
+        if let Some(exception_param) = &catch_clause.param {
```
Comment (aapoalas) [inline nova_vm/src/engine/bytecode/executable.rs:1177]: **question:** Any idea how we should handle exiting declarative environments caused by throwing?

eg.
```js
try {
  if (foo) { // <-- this env needs to be popped off
    let a = null;
    null.maybeItWorks();
  }
} catch(e) {
}
```

**Review by aapoalas — APPROVED** [review]
Absolutely splendid! I had one question about how declaration environment popping should work with throws.

File: nova_vm/src/engine/bytecode/executable.rs (hunk 1 above)
Comment (andreubotella) [reply nova_vm/src/engine/bytecode/executable.rs:1177]: I hadn't taken that into account. I guess exception jump targets could store their declarative environment, and that would be restored when an exception is thrown.

Comment (aapoalas) [reply nova_vm/src/engine/bytecode/executable.rs:1177]: Oh yeah, that would work and make a lot of sense. So `PushExceptionJumpTarget` and `PopExceptionJumpTarget` should push/pop the jump target together with the parent environment to restore when an exception is thrown.

**Pull request merged**